### DISCUSSION TITLE
Q&A: Low-carbon accessible project templates

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/gsoc#122] [General]
👋 Please use this discussion thread to ask questions about the [Low-carbon accessible project templates](https://github.com/wagtail/gsoc/blob/main/project-ideas.md#low-carbon-accessible-project-templates) project idea. 

## Asking questions

Please use this discussion thread.

## Proposals

It’s at mentors’ discretion whether they review any draft proposals or only final ones. To send your proposal for draft review (no promises), use: <https://wagtail.org/gsoc-proposal/>. We don’t mandate any specific template but we do provide one **optionally** as a way to get started: <https://wagtail.org/gsoc-template/>

---

For further information about the project, see:

- [GSoC 2023 sustainability-focused projects](https://wagtail.org/blog/wagtail-cms-projects-for-google-summer-of-code-2023/)
- [Estimating Wagtail websites’ emissions](https://wagtail.org/blog/estimating-wagtail-websites-emissions/)
- [RFC 90: sustainability roadmap](https://github.com/wagtail/rfcs/pull/90)

**Comment by yunchipang** [comment]
hello, i'm a django developer who recently discovered wagtail and am interested in contributing to the project through GSoC 2024✨ I went through all the docs (hopefully🤞) about a greener wagtail template and would love to write a proposal on this idea!

based on the description in [project-ideas](https://github.com/wagtail/rfcs/pull/90), does the anticipated outcome resemble the [bakerydemo](https://github.com/wagtail/bakerydemo) or is it closer to the style of the [getting started](https://docs.wagtail.org/en/stable/getting_started/tutorial.html) documentation? or neither? what i'm imagining now, is a wagtail template with default settings (either through static site hosting, scale-to-zero backends, and/or hybrid client/server rendering) that lead to lower carbon emissions, and users can use that as the starter template for their greener website. but i'm not sure if this is what the team's looking for.

any guidance from the mentors will be truly helpful, as i'm trying to understand what the team aims to achieve through this task. thank you 🙏🏻

**Reply by thibaudcolas** [reply]
Good question @yunchipang! Yep, I think somewhere in-between the bakerydemo and our "getting started" tutorial. Enough that you can use the template and get a site that works in a few commands, and not too much that you’d need to remove most of what the template comes with right after.

And yes, "a template with default settings" is what we’re after. So there’s the hosting options you mentioned, that’s a must, and then probably demonstrating some other ways to make Wagtail websites lighter. For example [light YouTube embeds](https://github.com/RealOrangeOne/wagtail-lite-youtube-embed).

**Comment by nirban07** [comment]
Hi, this is my understanding of the project till now. We are trying to reduce the carbon emissions that are made by wagtail pages by reducing the size of the files transferred in the network. We are focused on achieving this by reducing the image sizes by AVIF and WEBP Image Support. 
i would be grateful if you would let us know were we are on the image support and emission benchmarks so we can contribute without getting in anyones way.
thank you.

**Reply by thibaudcolas** [reply]
Hi @nirban07, I wouldn’t worry too much about getting in anyone’s way as far as making a proposal. We want proposals to demonstrate your understanding of Wagtail as a project and of the problem space, but we don’t expect full context on what’s currently in progress / very recent work. We’ll adjust the project plans once we get going anyway.

**Comment by rohitsrma** [comment]
Hello @thibaudcolas, Will these templates be similar to the [wagtail-cookiecutter template](https://github.com/chrisdev/wagtail-cookiecutter-foundation) but with a greater emphasis on sustainability?

**Reply by thibaudcolas** [reply]
With the caveat that I don’t know what `wagtail-cookiecutter` project you’re referring to… yes? Similar as in they will be used to bootstrap new websites. Different because we want different technologies, and yes this focus on low-footprint websites.

**Reply by rohitsrma** [reply]
Understood! Sorry for forgetting to include the link. It's now been added. It is built using wagtail and Zurb Foundation framework.

**Reply by thibaudcolas** [reply]
Yes, similar to that :) I think a good initial step of the project will be to try out some of the existing templates like `wagtail-cookiecutter-foundation` or [Pipit](https://github.com/Frojd/Wagtail-Pipit) to see what they offer.

Also worth noting Wagtail’s built-in project templating is based on what’s built into Django (see Django [TemplateCommand](https://github.com/django/django/blob/main/django/core/management/templates.py), not cookiecutter. I don’t think it matters too much but we’d need to figure out which option to use.

**Reply by rohitsrma** [reply]
Ok, Thank you so much!!

**Comment by nandini584** [comment]
Concerning this project,  I wanted to know about benchmarking tools to use for evaluating carbon emission.
I also wanted to know what type of templates we are planning to build, in the proposal shall I mention the ways to reduce carbon emission for templates to be made? Also about the tech stack, shall I only use the mentioned tech stacks, do I select specific ones or i have to use all according to the preferences?

**Reply by thibaudcolas** [reply]
> benchmarking tools to use for evaluating carbon emission.

There’s lots available in this space. The ones I’d recommend personally are [Green Metrics Tool](https://www.green-coding.io/projects/green-metrics-tool/), [GreenFrame](https://greenframe.io/), [Cloud Carbon Footprint](https://www.cloudcarbonfootprint.org/), and tools that look at the weight of web pages only like [Website Carbon](https://www.websitecarbon.com/).

For the purpose of a successful GSoC proposal, I don’t think it’s necessary to have deep knowledge of any one of those tools. Even if you knew about benchmarking for performance, that’d go a long way.

> I also wanted to know what type of templates we are planning to build, in the proposal shall I mention the ways to reduce carbon emission for templates to be made?

Not sure I understand what kind of info you’re after here but I’ll give it a go 😄 we want templates that are a solid basis to create Wagtail sites with, that demonstrate some of Wagtail’s features, and that do so in a way that reduce websites’ carbon footprint.

> shall I only use the mentioned tech stacks, do I select specific ones or i have to use all according to the preferences?

We’re happy with any tech stack in the proposal, though I’d recommend focusing on the ones that you’re personally familiar with and are reasonably popular and definitely have the potential to reduce carbon emissions. Good examples would be "Wagtail + static site generator", or "Wagtail + modern JS framework". Or pure "Wagtail" and no other tech, but innovative hosting setup (fully static hosting, or serverless Wagtail, or "Wagtail w/ SQLite database").

The "desirable" ones I mentioned are just based on the expertise of our mentors, other options very welcome.

**Reply by nandini584** [reply]
Thankyou so much, I think I get the clarity now.

**Comment by Abhinavcode13** [comment]
@thibaudcolas Can you clarify what kind of templates are you looking for if you have any such example?

**Reply by thibaudcolas** [reply]
I’ll try to produce an example now so it’s clearer

**Reply by thibaudcolas** [reply]
Hi @Abhinavcode13, here’s an example: https://github.com/thibaudcolas/benchmark-wagtail-demo

I chose to not demonstrate the "scaffolding" aspects of creating a project template, and instead focus on demonstrating what a site created with such a template would contain, as well as benchmarking the carbon footprint.

If you want a Wagtail project template example, here’s an existing one which is the results of our official tutorial: https://github.com/thibaudcolas/wagtail-tutorial-template.

**Reply by Abhinavcode13** [reply]
Ah great this would definitely help me @thibaudcolas , not sure if I can get my proposal submitted on time but I'll give it a try.

**Comment by nandini584** [comment]
@thibaudcolas Is it possible for me to get my proposal reviewed by you to make improvements if required?

**Reply by Abhinavcode13** [reply]
Too late I guess ;( @nandini584 anyways you can try.

**Reply by thibaudcolas** [reply]
Too late indeed :[ We’re reviewing final proposals now, and Google will announce the results on May 1st. Thank you for giving this a go!

**Comment by thibaudcolas** [comment]
Just wanted to say _thank you_ to everyone who submitted a proposal for this project! We received 7 in total, which is much more than I would have expected, and will certainly make for some good competition.

From here – we’ll be firming up our line-up of mentors, and reviewing all proposals. Final results will be announced by Google on [May 1st at 18:00 UTC](https://developers.google.com/open-source/gsoc/timeline#may_1_-_1800_utc).
